src/hooks/logging.rs

```diff
@@ -0,0 +1,45 @@
+use crate::{EvaluationContext, EvaluationDetails, EvaluationError, Value};
+
+use super::{Hook, HookContext, HookHints};
+
+/// A hook that logs the evaluation lifecycle of a flag.
+pub struct LoggingHook;
+
+#[async_trait::async_trait]
+impl Hook for LoggingHook {
+    async fn before<'a>(
+        &self,
+        context: &HookContext<'a>,
+        _: Option<&'a HookHints>,
+    ) -> Result<Option<EvaluationContext>, EvaluationError> {
+        log::debug!("Before hook for flag {}", context.flag_key);
+
+        Ok(None)
+    }
+    async fn after<'a>(
+        &self,
+        context: &HookContext<'a>,
+        _: &EvaluationDetails<Value>,
+        _: Option<&'a HookHints>,
+    ) -> Result<(), EvaluationError> {
+        log::debug!("After hook for flag {}", context.flag_key);
```

Comment:
please consider adding fields (Logged data) to the log messages (serialized, opt-in) according to [the specification](https://github.com/open-feature/spec/blob/main/specification/appendix-a-included-utilities.md#logging-hook) in each Stage.